src/LibraryManager.Vsix/UI/Controls/Library.xaml.cs

```diff
@@ -87,20 +108,38 @@ private void Commit(Completion completion)
 
             Text = completion.CompletionItem.InsertionText;
             LibrarySearchBox.CaretIndex = Text.IndexOf(completion.CompletionItem.DisplayText, StringComparison.OrdinalIgnoreCase) + completion.CompletionItem.DisplayText.Length;
+            Flyout.IsOpen = false;
+            SelectedItem = null;
         }
 
         private void HandleKeyPress(object sender, KeyEventArgs e)
         {
             switch (e.Key)
             {
                 case Key.Tab:
-                case Key.Enter:
+                    // SelectedItem could be null if the key press came from keyboard navigation and not commit operation.
+                    // In this case we will just move the focus to next control.
+                    if (SelectedItem == null)
+                    {
+                        LibrarySearchBox.MoveFocus(new TraversalRequest(FocusNavigationDirection.Next));
+                        e.Handled = true;
+                        break;
+                    }
+
                     Commit(SelectedItem);
+                    LibrarySearchBox.Focus();
+                    e.Handled = true;
                     break;
-                case Key.Escape:
+                case Key.Enter:
+                    Commit(SelectedItem);
```

Comment:
>Commit(SelectedItem); [](start = 20, length = 21)

Can SelectedItem be null here?

Reply:
It could be null if someone just hits enter inside textbox. It is handled in Commit(..) appropriately. The reason I added it above is to make sure we handle keyboard navigation correctly. Let me know if it doesn't make sense.

---
In reply to: [200471769](https://github.com/aspnet/LibraryManager/pull/278#discussion_r200471769) [](ancestors = 200471769)